### PR TITLE
Fix soundness hole of forgotten reach capabilities

### LINKED ISSUE #20503: Reach capabilities of function arguments get ignored
## Compiler version

`main`

## Minimized code

```Scala
import language.experimental.captureChecking
def runOps(ops: List[() => Unit]): Unit =
  ops.foreach(op => op())

def main(): Unit =
  val f: List[() => Unit] -> Unit = runOps  // should be error
```

## Output

It compiles.

## Expectation

It should not, as `runOps` unboxes the list elements and executes them, so `runOps` should not be pure. Based on the theory, we should simply prevent `runOps` from typechecking.

### PR DESCRIPTION
**Pull request by odersky**
Fixes #20503

### CODE REVIEW

**Review by Linyxus — APPROVED** [review]
LGTM!

File: tests/neg/i20503.scala
Comment (hamzaremmal) [inline tests/neg/i20503.scala:1]: The following test failed in nightly with `Scala 2 libary CC TASTy`. 
```
Wrong number of errors encountered when compiling tests/neg/i20503.scala
expected: 1, actual: 2
Unfulfilled expectations:

Unexpected errors:
tests/neg/i20503.scala:3
-> following the errors:
 at 3: Reach capability cap and universal capability cap cannot both
appear in the type [U](f: box (() ->{ops*} Unit) => U): Unit of this expression
 at 6: Found:    (ops: List[box () ->? Unit]^?) => Unit
Required: (ops: List[box () => Unit]) -> Unit
```

See: https://github.com/scala/scala3/actions/runs/9377502987/job/25819168381

**Comment by odersky** [comment]
@noti0na1 @Linyxus Can one of you follow up on this please. I don't have the bandwidth right now.

**Comment by noti0na1** [comment]
I will look into this error

**Comment by odersky** [comment]
@noti0na1 Thank you!

**Comment by noti0na1** [comment]
The extra error is at the type of `foreach: [U](f: A => U): Unit`.

When compiled with `Scala2LibraryCCTasty` using the bootstrapped compiler, symbol `foreach` is considered already checked with CC, so it is not fluidified.

I tried the commit before this PR, and I can trigger the same error using the same setting. So this error is not related to this PR.

File: tests/neg/i20503.scala
Comment (hamzaremmal) [reply tests/neg/i20503.scala:1]: Fixed by #20528

**Pull request merged**